docs/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations_for_sqlalchemy.rst

```diff
@@ -3,16 +3,266 @@
+    
+    In the case of ``MetaSqlAlchemyDataset``, the column name string will be passed through to your custom function unaltered. From there, you can use sqlalchemy's ``sqlalchemy.column`` method to create a ``ColumnClause`` based on it. Your function must return a valid ``ColumnClause``, which the decorator will combine with other SqlAlchemy methods to covert your logic into valid SQL queries and wrap the results as an :ref:``ExpectationValiadationResult``.
+
+    By convention, ``column_map_expectations`` always start with ``expect_column_values_...`` or ``expect_column_value_...`` (Ex: ``expect_column_value_word_counts_to_be_between``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+    |
+
+    You can also define ``column_aggregate_expectations``, which generate a single observed value for a whole column and evaluate it against some criteria. Like ``column_map_expectations``, ``column_aggregate_expectations`` also take at least two arguments (``self`` and ``column``). In this case, it's your responsibility to create all the logic for the full query, then wrap the result in a return object with a Boolean ``success`` value and ``observed_value`` nested inside ``result``.
+    
+    .. code-block:: python
+
+        @MetaSqlAlchemyDataset.column_aggregate_expectation
+        def expect_column_mode_to_equal(self, column, value):
+            mode_query = sa.select([
+                sa.column(column).label('value'),
+                sa.func.count(sa.column(column)).label('frequency')
+            ]).select_from(self._table).group_by(sa.column(column)).order_by(sa.desc(sa.column('frequency')))
+
+            mode = self.engine.execute(mode_query).scalar()
+            return {
+                "success": mode == value,
+                "result": {
+                    "observed_value": mode,
+                }
+            }
+
+    By convention, ``column_aggregate_expectations`` always start with ``expect_column_{property}_...`` (Ex: ``expect_column_mean_to_be_between``, ``expect_column_most_common_value_to_be_in_set``, ``expect_column_chisquare_test_p_value_to_be_greater_than``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+    |
+
+    Please see the :ref:`SqlAlchemyDataset` module for additional examples of how to contruct logic for Expectations. To learn about other Expectation types, please see :ref:`Other Expectation decorators` below.
+
+
+4. **Load some data**
+
+    To make your new Expectations available for validation, you can instantiate a ``CustomSqlAlchemyDataset`` as follows:
```

Comment:
Update to use Datasource, I think, again since that's where the Tutorial will leave you. (And someone who is creating custom expectations will have encountered a DataContext).

Reply:
My reasoning is the same as the other PR.